src/composables/secureStorageRef.ts

```diff
@@ -0,0 +1,137 @@
+import { Ref, ref, watch } from 'vue';
+import { validateMnemonic } from '@aeternity/bip39';
+
+import type { StorageKey } from '@/types';
+import { decrypt, encrypt } from '@/utils';
+import { IS_MOBILE_APP, STORAGE_KEYS } from '@/constants';
+import { WalletStorage } from '@/lib/WalletStorage';
+import { useAccounts } from './accounts';
+import { type ICreateStorageRefOptions, useStorageRef } from './storageRef';
+
+/**
+ * Create a secure storage ref that encrypts and decrypts the value using the password.
+ * Is a wrapper around the useStorageRef composable.
+ */
+export function useSecureStorageRef<T = string | object | any[]>(
+  initialState: T,
+  storageKey: StorageKey,
+  options: Omit<
+    ICreateStorageRefOptions<T>,
+    'serializer' | 'enableSecureStorage' | 'onBackgroundSync'
+  > = {},
+) {
+  async function getDecryptedValue(encryptedValue: any) {
+    if (IS_MOBILE_APP || !encryptedValue) {
+      return encryptedValue;
+    }
```

Comment:
without knowing the full logic, this feels weird to just return the same encrypted value in a function called  `getDecryptedValue` would expect this to be an error and caught elsewhere if needed